let User,CustomErrorHandler;_a49‍.x([["default",()=>_a49‍.o]]);_a49‍.w("../models",[["User",["User"],function(v){User=v}]]);_a49‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const admin = async (req, res, next) => {
    try {
        const user = await User.findOne({_id: req.user._id});

        if(user.role === 'admin'){
            next();
        }else{
            return next(CustomErrorHandler.unAuthorized());
        }
    } catch (error) {
        return next(CustomErrorHandler.serverError());
    }
}


_a49‍.d(admin);